File: Python/find-pivot-index.py
class Solution:
    def pivotIndex(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        totalSum = sum(nums)
        leftSum = 0

        for i, num in enumerate(nums):
            if leftSum == totalSum - leftSum - num:
                return i
            leftSum += num

        return -1

File: Swift/find-pivot-index.swift
class Solution {
    func pivotIndex(_ nums: [Int]) -> Int {
        let totalSum = nums.reduce(into: 0, { $0 += $1 })
        var leftSum = 0

        for (i, num) in nums.enumerated() {
            if leftSum == totalSum - leftSum - num {
                return i
            }
            leftSum += num
        }

        return -1
    }
}
